llpc/translator/lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -911,7 +911,7 @@ class SPIRVPhi : public SPIRVInstruction {
     assert(WordCount == Pairs.size() + FixedWordCount);
     assert(OpCode == OC);
     assert(Pairs.size() % 2 == 0);
-    foreachPair([=, this](SPIRVValue *IncomingV, SPIRVBasicBlock *IncomingBB) {
+    foreachPair([&](SPIRVValue *IncomingV, SPIRVBasicBlock *IncomingBB) {
```

Comment:
Is `validate` only called in debug mode?

Otherwise, we might want to wrap this call to `foreachPair` into `#ifndef NDEBUG` since it constructs a `std::function` for the callback, so I'd be surprised if the compiler can optimize it away.

Reply:
validate is also called in release mode